core/drivers/gpio/gpio.c

```diff
@@ -29,41 +32,33 @@ TEE_Result gpio_dt_alloc_pin(struct dt_pargs *pargs, struct gpio **out_gpio)
 	return TEE_SUCCESS;
 }
 
-static char *gpio_get_dt_prop_name(const char *gpio_name)
-{
-	int ret = 0;
-	char *prop_name = NULL;
-	size_t max_len = strlen(gpio_name) + strlen("-gpios") + 1;
-
-	prop_name = calloc(1, max_len);
-	if (!prop_name)
-		return NULL;
-
-	ret = snprintf(prop_name, max_len, "%s-gpios", gpio_name);
-	if (ret < 0 || (size_t)ret >= max_len) {
-		free(prop_name);
-		return NULL;
-	}
-
-	return prop_name;
-}
-
 TEE_Result gpio_dt_get_by_index(const void *fdt, int nodeoffset,
 				unsigned int index, const char *gpio_name,
 				struct gpio **gpio)
 {
 	TEE_Result res = TEE_ERROR_GENERIC;
-	char *prop_name = NULL;
+	char prop_name[32]; /* 32 is max size of property name in DT */
 	void *out_gpio = NULL;
+	unsigned int i = 0;
 
-	prop_name = gpio_get_dt_prop_name(gpio_name);
-	if (!prop_name)
-		return TEE_ERROR_OUT_OF_MEMORY;
+	/* Try GPIO properties "foo-gpios" and "foo-gpio" */
+	for (i = 0; i < ARRAY_SIZE(gpio_suffixes); i++) {
+		if (gpio_name)
+			snprintf(prop_name, sizeof(prop_name), "%s-%s",
+				 gpio_name, gpio_suffixes[i]);
```

Comment:
`snprintf()` return value to check?  I could not find where in the DT spec property names shall be at most 32bytes (including terminal \0 char?)

Reply:
The limit of 32 bytes depends on the devicetree specifications : 
https://devicetree-specification.readthedocs.io/en/v0.3/devicetree-basics.html#properties
[2.2.4.1](http://2.2.4.1/). 
In linux the function of_find_gpio() has the following comment : 
`char propname[32]; /* 32 is max size of property name */`

Reply:
Ok, clear. Thanks.

For consistency and help maintenance, could you introduce a `DT_PROP_NAME_SIZE_MAX` macro in **core/kernel/dt.h** through a dedicated commit (which commit message could refer to URL you pointed above)?

I still think `snprintf()` return value should be checked, at least asserted.

Reply:
You're right. Checking the return value is not needed.
Discard my comment. Sorry for the noise.